code/game/objects/structures/mortar.dm

```diff
@@ -9,109 +9,110 @@
-		if(temp_dial_x + targ_x > world.maxx || temp_dial_x + targ_x < 0)
-			to_chat(user, "<span class='warning'>You cannot dial to this coordinate, it is outside of the area of operations.</span>")
-			return
-		if(temp_dial_x < -10 || temp_dial_x > 10)
-			to_chat(user, "<span class='warning'>You cannot dial to this coordinate, it is too far away. You need to set [src] up instead.</span>")
-			return
-		var/temp_dial_y = input("Set latitude adjustement from -10 to 10.") as num
-		if(temp_dial_y + targ_y > world.maxy || temp_dial_y + targ_y < 0)
-			to_chat(user, "<span class='warning'>You cannot dial to this coordinate, it is outside of the area of operations.</span>")
-			return
-		var/turf/T = locate(targ_x + temp_dial_x, targ_y + temp_dial_y, z)
-		if(get_dist(loc, T) < 10)
-			to_chat(user, "<span class='warning'>You cannot dial to this coordinate, it is too close to your mortar.</span>")
-			return
-		if(temp_dial_y < -10 || temp_dial_y > 10)
-			to_chat(user, "<span class='warning'>You cannot dial to this coordinate, it is too far away. You need to set [src] up instead.</span>")
-			return
-		if(busy)
-			to_chat(user, "<span class='warning'>Someone else is currently using this mortar.</span>")
+	if(user.skills.getRating("engineer") < SKILL_ENGINEER_ENGI)
+		user.visible_message("<span class='notice'>[user] fumbles around figuring out how to use [src].</span>",
+		"<span class='notice'>You fumble around figuring out how to use [src].</span>")
+		var/fumbling_time = 4 SECONDS * ( SKILL_ENGINEER_ENGI - user.skills.getRating("engineer") )
+		if(!do_after(user, fumbling_time, TRUE, src, BUSY_ICON_UNSKILLED))
 			return
-		user.visible_message("<span class='notice'>[user] starts dialing [src]'s firing angle and distance.</span>",
-		"<span class='notice'>You start dialing [src]'s firing angle and distance to match the new coordinates.</span>")
-		busy = 1
-		playsound(loc, 'sound/items/ratchet.ogg', 25, 1)
-		if(do_after(user, 15, TRUE, src, BUSY_ICON_GENERIC))
-			user.visible_message("<span class='notice'>[user] finishes dialing [src]'s firing angle and distance.</span>",
-			"<span class='notice'>You finish dialing [src]'s firing angle and distance to match the new coordinates.</span>")
-			busy = 0
-			dial_x = temp_dial_x
-			dial_y = temp_dial_y
-		else busy = 0
+	ui_interact(user)
+
 
+/obj/structure/mortar/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = 0, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.default_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "Mortar", "M402 Mortar", 430, 150, master_ui, state)
+		ui.open()
+
+/obj/structure/mortar/ui_data(mob/user)
+	. = ..()
+	var/data = list()
+	data["X"] = coords["targ_x"]
+	data["Y"] = coords["targ_y"]
+	data["D1"] = coords["dial_x"]
+	data["D2"] = coords["dial_y"]
+	data["last_three_inputs"] = last_three_inputs
+	return data
```

Comment:
```suggestion
/obj/structure/mortar/ui_data(mob/user)
	. = list()
	.["X"] = coords["targ_x"]
	.["Y"] = coords["targ_y"]
	.["D1"] = coords["dial_x"]
	.["D2"] = coords["dial_y"]
	.["last_three_inputs"] = last_three_inputs
```
also rename D1 and D2 to something more like Dialx and Dialy

Reply:
Ill do that